docs/portingguide/src/howto/nogc.md

```diff
@@ -1,50 +1,283 @@
+
+## Allocation
+Now we can finally implement the allocation functions.
+
+### Runtime changes
+Add the following two functions to the `mmtk.h` file:
+
+```C
+[...]
+
+/**
+ * Allocate an object
+ *
+ * @param mutator the mutator instance that is requesting the allocation
+ * @param size the size of the requested object
+ * @param align the alignment requirement for the object
+ * @param offset the allocation offset for the object
+ * @param allocator the allocation semantics to use for the allocation
+ * @return the address of the newly allocated object
+ */
+void *mmtk_alloc(MmtkMutator mutator, size_t size, size_t align,
+        ssize_t offset, int allocator);
+
+/**
+ * Set relevant object metadata
+ *
+ * @param mutator the mutator instance that is requesting the allocation
+ * @param object the returned address of the allocated object
+ * @param size the size of the allocated object
+ * @param allocator the allocation semantics to use for the allocation
+ */
+void mmtk_post_alloc(MmtkMutator mutator, void* object, size_t size, int allocator);
+
+[...]
+```
+
+In order to perform allocations, you will need to know what object alignment the runtime expects. Runtimes often align allocations at word boundaries (i.e. 4- or 8-bytes) as it allows the CPU to access the data faster at execution time. Additionally, the language may use the unused lowest order bits to store flags (e.g. type information), so it is important that MMTk respects these expectations. Once you have figured out the alignment requirements for your runtime, you should update the [`VM::MIN_ALIGNMENT`](https://www.mmtk.io/mmtk-core/public-doc/vm/trait.VMBinding.html#associatedconstant.MIN_ALIGNMENT) constant in `VMBinding` to the correct value.
+
+Now that MMTk is aware of each mutator thread, you have to change the runtime's allocation functions to call into MMTk to allocate using `mmtk_alloc` and set object metadata using `mmtk_post_alloc`. Note that there may be multiple allocation functions in the runtime so make sure that you edit them all!
+
+You should use the saved `Mutator` pointer as the first parameter, the requested object size as the next parameter, and any alignment requirements the runtimes has as the third parameter.
+
+If your runtime requires a non-zero allocation offset (i.e. returned address vs actual allocated address) then you have to provide the required value as the fourth parameter. Note that you ***must*** also update the [`USE_ALLOCATION_OFFSET`](https://www.mmtk.io/mmtk-core/public-doc/vm/trait.VMBinding.html#associatedconstant.USE_ALLOCATION_OFFSET) constant in the `VMBinding` implementation if your runtime requires a non-zero allocation offset.
```

Comment:
> a non-zero allocation offset (i.e. returned address vs actual allocated address)

This is confusing. `offset` menas we make sure alignment for **the returned address at the offset**, instead of alignment for **the returned address**.

Reply:
This isn't immediately clear from the `alloc` function's documentation. I will fix it in the porting guide, but it needs to be fixed in the doc comment as well.

Reply:
You can just add that to the comments in this PR. Yeah, it is not clear.